src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts

```diff
@@ -0,0 +1,394 @@
+      )
+    );
+
+    const isNewValue = (
+      fuzzyDate !== this._value ||
+      !areFuzzyDatesEqual
+    );
+
+    this._value = fuzzyDate || value;
+
+    if (isNewValue) {
+      this.onChange(this._value);
+
+      // Do not mark the field as "dirty"
+      // if the field has been initialized with a value.
+      if (this.isFirstChange && this.control) {
+        this.control.markAsPristine();
+      }
+
+      if (this.isFirstChange && this._value) {
+        this.isFirstChange = false;
+      }
+
+      this.datepickerComponent.selectedDate = dateValue;
+    }
+
+    this.setInputElementValue(formattedDate || '');
+  }
+
+  constructor(
+    protected changeDetector: ChangeDetectorRef,
+    protected configService: SkyDatepickerConfigService,
+    protected elementRef: ElementRef,
+    protected renderer: Renderer2,
+    protected resourcesService: SkyLibResourcesService,
+    @Optional() protected datepickerComponent: SkyDatepickerComponent,
+    private fuzzyDateFactory: SkyFuzzyDateFactory
+  ) {
+    super(
+      changeDetector, configService, elementRef,
+      renderer, resourcesService, datepickerComponent);
+   }
+
+  public ngOnInit(): void {
+    if (this.yearRequired) {
+      if (this.dateFormat.toLowerCase().indexOf('y') === -1) {
+        throw new Error('You have configured conflicting settings. Year is required and dateFormat does not include year.');
+      }
+    }
+
+    if (!this.datepickerComponent) {
+      throw new Error(
+        'You must wrap the `skyFuzzyDatepickerInput` directive within a ' +
+        '`<sky-datepicker>` component!'
+      );
+    }
+
+    const element = this.elementRef.nativeElement;
+
+    this.renderer.addClass(
+      element,
+      'sky-form-control'
+    );
+
+    const hasAriaLabel = element.getAttribute('aria-label');
+
+    if (!hasAriaLabel) {
+      this.resourcesService.getString('skyux_date_field_default_label')
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe((value: string) => {
+          this.renderer.setAttribute(
+            element,
+            'aria-label',
+            value
+          );
+        });
+    }
+  }
+
+  public validate(control: AbstractControl): ValidationErrors {
+    if (!this.control) {
+      this.control = control;
+    }
+
+    if (this.skyDatepickerNoValidate) {
+      return;
+    }
+
+    if (!this.control.value) {
+      return;
+    }
+
+    const value: any = control.value;
+
+    let fuzzyDate: SkyFuzzyDate;
+    let isFuzzyDateValid = true;
+    let validationErrorMessage: string;
+
+    if (typeof value === 'string') {
+      fuzzyDate = this.fuzzyDateFactory.getFuzzyDateFromDateString(value, this.dateFormat);
+    } else {
+      fuzzyDate = value;
+    }
+
+    if (!fuzzyDate) {
+      if (this.dateFormatErrorMessage) {
+        validationErrorMessage = this.dateFormatErrorMessage;
+      } else {
+        this.resourcesService.getString('skyux_fuzzy_datepicker_invalid_format')
```

Comment:
We generally don’t set any error messages in the `validate` method of `ControlValueAccessor`; instead, we’d create a unique "slug" that would identify the error and let that bubble up to the consumer. They would then detect the specific error and write out the message themselves. For example:

```
return {
  skyFuzyDate: {
    invalidFormat: true,
    yearRequired: true,
    // etc.
  }
};
```